Added options to customize the MaterialPreviewOverlay

job-46 opened the pull request:
First of all thanks for this package!
Finally a normal and modern working QR and barcode scanner.

I have added some options to customize the scanner view to your app's look and feel.
- Option for a square scan area (handy for if you only scan QR Codes)
- Option to change the scan view border-color
- Option to change the scan background-color

If there is something wrong let me know!

job-46 committed:
Added options to customize the MaterialPreviewOverlay

- Option for a square scan area (handy for if you only scan QR Codes)
- Option to change the scan view border color
- Option to change the scan background color

job-46 committed:
Small performance fix

Changed the order of the cutOutHeight. Now it assumes a square by default (so without the calculation). If it should not be a square, it does the calculation.

jhoogstraat reviewed (commented):
Glad that you like it! I had the same thought and just built it myself. I am always happy for new contributions.

Isn't `squareScanView` the same as `aspectRatio = 1`? Why the additional paramter?

JobMoll commented:
That's right! But I thought a bool was easier than calculating it yourself.
But aspectRatio is more customizable by the user.

In any case, indeed I should remove aspectRatio or the squareScanView. 
If you tell me what you'd like more I will fix the code 👍

jhoogstraat commented:
Thinking about it, I guess nobody will ever use an aspect ratio other than 16/9 or 1/1.

What do you think about an enum `AspectRatio` with two cases `square` and `wide`?

job-46 committed:
Removed squareScanView

- removed squareScanView
- aspectRatio to scanViewAspectRatio to make it more clear.

job-46 commented:
Ah that's even better yea! Ignore my previous commit then xD

job-46 committed:
Added enum ScanViewShape

job-46 commented:
Here you go.

I have changed AspectRatio to ScanViewShape because I think it is clearer what it does.

jhoogstraat commented:
Yeah, the naming was bad. But it is also not a view, but more of a scannable area or cut out. So maybe `ScanArea` or `CutoutShape`? Something short.

job-46 commented:
I like the CutOutShape!

Something like this is sufficient:
scanViewBorderColor to cutOutShapeBorderColor.
scanViewShape to cutOutShape.
ScanViewShape.square & ScanViewShape.wide to CutOutShape.square & CutOutShape.wide.

job-46 committed:
ScanView to CutOut

jhoogstraat commented:
Great! Because `AspectRatio` is gone now, these are breaking changes, actually.
Thus, I have to merge this into the develop branch.

Can you rebase your changes?

job-46 commented:
Does this work? Or should I clone the dev branche first and then apply the changes there?

jhoogstraat commented:
Yes, you can rebase your changes on top of the develop branch and fix the merge conflicts (locally) and then force push to your remote branch. I will squash your changes anyway.
